RELEASE.md

```diff
@@ -0,0 +1,101 @@
+  - Label the PR: `status: cherry-picked 🍒`.
+
+_Outcome_: __Release branch has all relevant changes merged & pushed.__
+
+### Prepare release
+#### Ensure release branch readme is up to date
+- Run changelog script `npm run changelog` to get changelog txt for readme. Changelog content will be output to screen by script.
+- Add changelog section for release, e.g. `= 2.5.11 - 2020-01-20 =`.
+- Copy-paste the changelog content into `readme.txt`.
+- Make any other changes to readme as needed - e.g. support versions changing, new blocks. 
+- Push readme changes to release branch on origin repo.
+
+_Outcome_: __`readme.txt` is up to date with changes for release.__
+
+#### Build zip & smoke test
+- Ensure you are on tip of release branch.
+- Update dependencies – `npm ci`.
+- Run a production build - `npm run build`.
+- Run package script to get a zip to test `npm run package-plugin`.
+- Smoke test built release zip:
+  - Recommend doing this in a clean environment, e.g. Jurassic.Ninja site.
+  - Sanity check - check correct blocks exist and function.
+  - Test to confirm new features/fixes are working correctly.
+  - Smoke test – test a cross section of core functionality.
+
+_Outcome_: __Confident that source code is ready for release: intended fixes are working correctly, no release blockers or build issues.__
+
+### Release!
+#### Release to GitHub
+- Prepare tagged release on github `npm run deploy`. 
+  - Note: the script automatically updates version numbers (commits on your behalf).
+- Edit release, add changelog info to Github release notes.
+- Check release repo tag is correct - checkout, smoke test/sanity check.
+
+_Outcomes_: __Version numbers updated in source code & developers can test tagged release.__
+
+##### Release to WPORG
+- Push release to WPORG `npm run release`.
+- Note that this may happen a day or so after Github release.
+- FYI: Sometimes we make small changes (e.g. readme only) by [committing directly to svn repo](https://developer.wordpress.org/plugins/wordpress-org/how-to-use-subversion/#editing-existing-files).
+
+_Outcome_: __Customers can install/update via WPORG; WPORG plugin page is up to date__.
+
+### After release
+#### Update `master` with release changes
+- Ensure changelog is up to date on master.
+- If major/minor release, update version on master with dev suffix, e.g. [`2.6-dev`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/commit/e27f053e7be0bf7c1d376f5bdb9d9999190ce158).
+- Code changes should already be present on master - all dev happens on `master`.
+
+
+## `@todo` Missing info to link to / document
+- Release milestones
+- Relationship to Woo core release
+- Testing blocks plugin as included in Woo core (do we do this?)
+- Policy for inclusion in major / minor / fix releases - i.e. how we use versioning 
+- ["Legacy"](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/add/release-process/assets/js/legacy/README.md) stuff - is this relevant to releases?
```

Comment:
The legacy stuff is handled by the build process so there's nothing specific the person doing the release has to do. However this does raise the point that as long as we still officially support WP 5.0, we should be testing against that version of WordPress to make sure the blocks work as expected (I think we often forget that step). Currently on versions of WP < 5.3, All Products, and the filter blocks won't be available there, but all our other "legacy" blocks should still work fine.

Reply:
Thanks for reminding me about this – I didn't quite understand yesterday :) 

I've added this to the pre-release final testing of the zip bbba14ded3a23d12bdf8437c0c9dd6c8cff88eb5:

`Test to confirm blocks are available and work correctly in oldest supported WordPress version (e.g. 5.0).`